app/src/main/java/com/besome/sketch/design/DesignActivity.java

```diff
@@ -216,22 +225,63 @@ private void generateProjectDebugFiles() {
         q.b(jC.b(sc_id), jC.a(sc_id), jC.c(sc_id));
     }
 
+    /** Open another app.
+     * @param context current Context, like Activity, App, or Service
+     * @param packageName the full package name of the app to open
+     * @return true if likely successful, false if unsuccessful
+     */
+    public static boolean openApp(Context context, String packageName) {
+        PackageManager manager = context.getPackageManager();
+        try {
+            Intent i = manager.getLaunchIntentForPackage(packageName);
+            if (i == null) {
+                return false;
+                //throw new ActivityNotFoundException();
+            }
+            i.addCategory(Intent.CATEGORY_LAUNCHER);
+            context.startActivity(i);
+            return true;
+        } catch (ActivityNotFoundException e) {
+            return false;
+        }
+    }
+
     /**
      * Opens the debug APK to install.
      */
-    private void installBuiltApk() {
-        Intent intent = new Intent(Intent.ACTION_VIEW);
-        if (Build.VERSION.SDK_INT >= 24) {
-            Uri apkUri = FileProvider.a(getApplicationContext(), getApplicationContext().getPackageName() + ".provider", new File(q.finalToInstallApkPath));
-            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
-            intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
-            intent.addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
-            intent.setDataAndType(apkUri, "application/vnd.android.package-archive");
-        } else {
-            intent.setDataAndType(Uri.fromFile(new File(q.finalToInstallApkPath)), "application/vnd.android.package-archive");
-        }
+    private void installBuiltApk() { // TODO
```

Comment:
Again, what's to do here?

Reply:
> Again, what's to do here?

I planned to make a ProgressDialog here, but decided to leave it for later, since AsyncTask causes errors

Reply:
Weird. Anyway, next time, please remove those comments if you didn't implement an idea.